### PR TITLE
Fixes #54 add batch domain to core

### LINKED ISSUE #54: Add Batch domain to core
This will be used in PKSim and in the OSPSuite.InstallationValidator tool

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
Moved objects from PKSim to Core. Also added some of the constants that you had marked (see other comments)

File: src/OSPSuite.Core/Domain/Constants.cs [hunk 1]
```diff
@@ -176,6 +176,14 @@ public static string ParameterCompoundType(int index)
          //end of  delete
       }
 
+      public static class RegistryPaths
+      {
+         public const string PKSIM_REG_PATH = @"Open Systems Pharmacology\PK-Sim\";
+         public const string MOBI_REG_PATH = @"Open Systems Pharmacology\MoBi\";
+         public const string INSTALL_DIR = "InstallDir";
+         public const string INSTALL_PATH = "InstallPath";
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Constants.cs]: This is actually the constant that you want to read in the registry. This gives you the PATH where PK-Sim is installed

**Comment by msevestre** [comment]
I don't get it. The build is stuck once all the tests are done. Going to have to investigate what's going on

File: tests/OSPSuite.Core.Tests/Batch/SimulationResultsExporterSpecs.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      {
+         base.Context();
+         var dataTable = new DataTable("TEST");
+
+         _dataTables = new List<DataTable> {dataTable};
+         A.CallTo(_dataRepositoryTask).WithReturnType<IEnumerable<DataTable>>().Returns(_dataTables);
+      }
+
+      [Observation]
+      public async Task should_create_a_data_table_with_all_results_from_the_simulation_and_export_it_to_the_file()
```
Comment (rwmcintosh) [inline tests/OSPSuite.Core.Tests/Batch/SimulationResultsExporterSpecs.cs]: Checking the appveyor logs here. This test appears gren in the list, but without a time.

File: tests/OSPSuite.Core.Tests/Batch/SimulationResultsExporterSpecs.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+      protected override void Context()
+      {
+         base.Context();
+         _batchSimulationExport = new BatchSimulationExport {Name = "Sim"};
+         A.CallTo(() => _batchSimulationExportMapper.MapFrom(_simulation, _results)).Returns(_batchSimulationExport);
+      }
+
+      [Observation]
+      public async Task should_create_a_batch_simulation_export_object_and_export_it_to_the_file()
```
Comment (rwmcintosh) [inline tests/OSPSuite.Core.Tests/Batch/SimulationResultsExporterSpecs.cs]: This test does not appear in the appveyor logs

File: tests/OSPSuite.Core.Tests/Batch/SimulationResultsExporterSpecs.cs (hunk 2 above)
Comment (rwmcintosh) [inline tests/OSPSuite.Core.Tests/Batch/SimulationResultsExporterSpecs.cs]: Actually, that test appears yellow.

**Comment by msevestre** [comment]
yeah I figured it would be an issue with some async tests. But why?!? Going to comment them out and see what happens

**Comment by msevestre** [comment]
Yeah using Wait instead of await does the trick. But I still don't get why...anyways

**Pull request merged**